### DISCUSSION TITLE
Cross Probing : Issue related to Bidirectional mapping.

### DISCUSSION THREAD

**Discussion by RameshwarShinde** [iTwin/itwinjs-core#4149] [Q&A] [answered]
Hi, 

With reference to [Cross Probing Example](https://www.itwinjs.org/sandboxes/iTwinPlatform/Cross-Probing), have implemented with my model. Able to map from 3d to 2d, but unable to map from 2d to 3d. 

As when I click on element in the 2d viewport, but getting error (in below snap)
Target element data is getting in console. I tried this with multiple iModels , still the issue is same  & not getting what's wrong. 
Is it related to iModel ?

![image](https://user-images.githubusercontent.com/74453268/186159301-e238c18e-ef5d-4a9e-b204-87e71bf967f3.png)

Can anyone plz help out to get me the bidirectional flow complete? Thanks in advance.

**Comment by pmconne** [comment, marked as answer]
I fixed this problem in [my fork of the sandbox](https://www.itwinjs.org/sandboxes/pmconne/Fixed-cross-probing). See `_get2DViewport` and `_get3DViewport` in CrossProbingApi.ts.

@Ellord207 when will the official example be fixed?

**Reply by Ellord207** [reply]
Should be out tomorrow.  I will update this thread when it is.

**Reply by RameshwarShinde** [reply]
Thanks @pmconne. Issue resolved. 
Just one question : Is cross probing possible with different iModels ? ( I mean two separate iModels)

**Reply by Ellord207** [reply]
Cross Probing Sample has been updated.
